### PR TITLE
Add missing urls for io types

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @steveklabnik

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
@frewsxcv orders so I execute.

@bors: delegate=kmcallister

File: src/libstd/io/cursor.rs [hunk 1]
```diff
@@ -15,22 +15,28 @@ use cmp;
...
-/// Cursors are typically used with in-memory buffers to allow them to
-/// implement `Read` and/or `Write`, allowing these buffers to be used
+/// `Cursor`s are typically used with in-memory buffers to allow them to
+/// implement [`Read`] and/or [`Write`], allowing these buffers to be used
 /// anywhere you might use a reader or writer that does actual I/O.
 ///
 /// The standard library implements some I/O traits on various types which
-/// are commonly used as a buffer, like `Cursor<Vec<u8>>` and `Cursor<&[u8]>`.
+/// are commonly used as a buffer, like `Cursor<`[`Vec`]`<u8>>` and
+/// `Cursor<`[`&[u8]`]`>`.
```
Comment (ollie27) [inline src/libstd/io/cursor.rs]: You can't put square brackets in links like this. It needs to be something like ``` `Cursor<`[`&[u8]`][bytes]`>`.``` and `[bytes]: ../../std/primitive.slice.html` for this link to work.

Comment (GuillaumeGomez) [inline src/libstd/io/cursor.rs]: Actually it does work.

Comment (ollie27) [inline src/libstd/io/cursor.rs]: I've fixed this as part of #37316.

**Pull request merged**